openedx/core/djangoapps/embargo/api.py

```diff
@@ -170,10 +170,15 @@ def _country_code_from_ip(ip_addr):
         str: A 2-letter country code.
 
     """
-    if ip_addr.find(':') >= 0:
-        return pygeoip.GeoIP(settings.GEOIPV6_PATH).country_code_by_addr(ip_addr)
-    else:
-        return pygeoip.GeoIP(settings.GEOIP_PATH).country_code_by_addr(ip_addr)
+    reader = geoip2.database.Reader(settings.GEOIP_PATH)
+
+    try:
+        response = reader.country(ip_addr)
+        country_code = response.country.iso_code
+    except geoip2.errors.AddressNotFoundError:
+        country_code = "unknown"
+    reader.close()
```

Comment:
Close reader inside try block.

Reply:
I don't understand why the close() should be in the try.  I assume it should be called even if the reader.country() call throws an exception.

Reply:
Sorry, my bad. You're right, it should be called even it throws exception.